modules/core/include/opencv2/core/dualquaternion.hpp

```diff
@@ -0,0 +1,857 @@
+     * @param moment the moment of line.
+     * @note Translation is applied after the rotation. Use createFromQuat(r, r * t / 2) to create
+     * a dual quaternion which translation is applied before rotation.
+     */
+    static DualQuat<_Tp> createFromPitch(const _Tp angle, const _Tp d, const Quat<_Tp> &axis, const Quat<_Tp> &moment);
+
+    /**
+     * @brief return a quaternion which represent the real part of dual quaternion.
+     * The definition of real part is in createFromQuat().
+     * @sa createFromQuat, getDualQuat
+     */
+    Quat<_Tp> getRealQuat() const;
+
+    /**
+     * @brief return a quaternion which represent the dual part of dual quaternion.
+     * The definition of dual part is in createFromQuat().
+     * @sa createFromQuat, getRealQuat
+     */
+    Quat<_Tp> getDualQuat() const;
+
+    /**
+     * @brief return the conjugate of a dual quaternion.
+     * \f[
+     * \begin{equation}
+     * \begin{split}
+     * \sigma^* &= (p + \epsilon q)^*
+     *          &= (p^* + \epsilon q^*)
+     * \end{split}
+     * \end{equation}
+     * \f]
+     */
+    DualQuat<_Tp> conjugate() const;
+
+    /**
+     * @brief return the rotation in quaternion form.
+     */
+    Quat<_Tp> getRotation(QuatAssumeType assumeUnit=QUAT_ASSUME_NOT_UNIT) const;
+
+    /**
+     * @brief return the translation in pure quaternion form.
+     * The rotation \f$r\f$ in this dual quaternion \f$\sigma\f$ is applied before translation \f$t\f$,
+     * which has the form
+     * \f[\begin{equation}
+     * \begin{split}
+     * \sigma &= p + \epsilon q \\
+     *        &= r + \frac{\epsilon}{2}\boldsymbol{t}r.
+     * \end{split}
+     * \end{equation}\f]
+     * Thus, the translation can be obtained as:
+     * \f[\boldsymbol{v} = 2qp^*.\f]
+     * @param assumeUnit if QUAT_ASSUME_UNIT, this dual quaternion assume to be a unit quaternion
+     * and this function will save some computations.
+     * @note this dual quaternion's translation is applied after the rotation.
+     */
+    Quat<_Tp> getTranslation(QuatAssumeType assumeUnit=QUAT_ASSUME_NOT_UNIT) const;
+
+    /**
+     * @brief return the norm \f$||A||\f$ of dual quaternion \f$A = p + \epsilon q\f$.
+     * \f[
+     *  \begin{equation}
+     *  \begin{split}
+     *  ||A|| &= \sqrt{A * A^*} \\
+     *        &= ||p|| + \epsilon \frac{p \cdot q}{||p||}.
+     *  \end{split}
+     *  \end{equation}
+     *  \f]
+     * Generally speaking, the norm of a not unit dual
+     * quaternion is a dual number. For convenience, we return it in the form of a dual quaternion
+     * , i.e.
+     * \f[ ||A|| = [||p||, 0, 0, 0, \frac{p \cdot q}{||p||}, 0, 0, 0].\f]
+     *
+     * @note The data type of dual number is dual quaternion.
+     * @param assumeUnit if QUAT_ASSUME_UNIT, this dual quaternion assume to be a unit quaternion and this f, this dual quaternion
+     */
+    DualQuat<_Tp> norm(QuatAssumeType assumeUnit=QUAT_ASSUME_NOT_UNIT) const;
```

Comment:
If the quaternion is assumed to be unit then its norm is (1, 0), so I guess this is an extra assumption.

Reply:
The norm of a unit dual quaternion is indeed (1,0).
But I'm sorry I didn't catch your meaning. What should I change here?

Reply:
I propose to remove `assumeUnit` argument from `norm()` function.
A user would pass `assumeUnit==true` only if they knows that this DQ has a norm of (1, 0) so he does not need to call this function at all.

Reply:
Yes, this is reasonable.